lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_dns
+short_description: Manage DNS configuration of an ESXI host system
+description:
+- This module can be used to configure DNS for ESXi host systems in given vCenter infrastructure.
+- It's only possible to configured DNS for the default TCP/IP stack.
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  host_name:
+    description:
+    - Host name
```

Comment:
The descriptions should be a bot more informative. Now they do not add any value.
- ```The short hostname to be used for the target``` (is it short, or fqdn here ?)
- The FQDN domain name.
- A list of DNS servers to be use.
- The order of the DNS servers is important as they are used consecutively in order.
- A list of domains to be searched through by the resolver.

Reply:
I think the FQDN is hostname plus domain name.

I added that information. Thanks.

P.S.: The PR was intended to be still WIP. I didn't expect that you will review everything now :)